### PR TITLE
Subs 1837

### PR DESCRIPTION
**Pull request by haseeb-gh**

### CODE REVIEW

File: src/docs/asciidoc/ref_file_upload.adoc [hunk 1]
```diff
@@ -75,3 +91,70 @@ include::{snippets}/delete-file/curl-request.adoc[]
...
+https://www.globus.org/get-started[getting started] guide for more information.
+
+== Requesting Globus share
+
+To begin uploading files, users will first need to request a Globus share. A share is basically a dedicated folder that
+DSP creates for the requesting user that they can upload their files into. The following REST API is used for requesting
+a share:
+
+=== Request
+include::{snippets}/create-globus-share/curl-request.adoc[]
```
Comment (ke4) [inline src/docs/asciidoc/ref_file_upload.adoc:107]: In the generated snippet there is a UUID. I think the user/reader of the doc would be confused what that UUID would be. We need to give some explanation, I think.

File: src/docs/asciidoc/ref_file_upload.adoc [hunk 2]
```diff
@@ -75,3 +91,70 @@ include::{snippets}/delete-file/curl-request.adoc[]
...
+
+=== Response
+include::{snippets}/create-globus-share/http-response.adoc[]
```
Comment (ke4) [inline src/docs/asciidoc/ref_file_upload.adoc:110]: In the response there is an origin_id field. Maybe we would need some description for that field.

File: src/docs/asciidoc/ref_file_upload.adoc [hunk 3]
```diff
@@ -75,3 +91,70 @@ include::{snippets}/delete-file/curl-request.adoc[]
...
+becomes available under <<ref_submissions.adoc#_get_a_submission,submission details>> should users wish to access it again later.
+
+== Notifying DSP about uploaded files
+
+Since Globus is an external third-party system, it is difficult for DSP to find out when a particular file has finished uploading.
+It is, therefore, the responsibility of the user to let DSP know whenever that happens. The following REST API
+acts as a way of notifying DSP about such files:
+
+=== Request
+include::{snippets}/notify-globus-uploadedfiles/curl-request.adoc[]
```
Comment (ke4) [inline src/docs/asciidoc/ref_file_upload.adoc:131]: UUID needs some explanation here too

File: src/docs/asciidoc/ref_file_upload.adoc [hunk 4]
```diff
@@ -75,3 +91,70 @@ include::{snippets}/delete-file/curl-request.adoc[]
...
+the files or not.
+
+NOTE: Calling the notify endpoint will remove the files from user's Globus directory and moves them to an internal staging area.
+This prevents the files from getting deleted by the users from Globus UI during or after data validation.
+Users are hereby advised not to get confused if they notice files disappearing from their Globus directory. If in doubt, use
+DSP's files API to see whether or not it appears in the list of files DSP has identified for a particular submission.
+
+== Share removal upon completion
+
+DSP automatically deletes the share as soon as it's last registered submission gets completed. If there are any files
```
Comment (ke4) [inline src/docs/asciidoc/ref_file_upload.adoc:148]: it's --> its

File: src/docs/asciidoc/ref_file_upload.adoc [hunk 5]
```diff
@@ -75,3 +91,70 @@ include::{snippets}/delete-file/curl-request.adoc[]
...
+present in the directory at that time then they are deleted as well.
```
Comment (ke4) [inline src/docs/asciidoc/ref_file_upload.adoc:149]: Are the users get any notification related to this possible file(s) deletion?

File: src/main/java/uk/ac/ebi/subs/api/config/MessagingConfig.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package uk.ac.ebi.subs.api.config;
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/api/config/MessagingConfig.java:1]: Do we need this class?

File: src/test/java/uk/ac/ebi/subs/api/documentation/SubmissionApiDocumentation.java [hunk 7]
```diff
@@ -166,9 +166,11 @@
 
     private String urlBase;
 
+    private Submission submission;
+
     @Before
     public void setUp() throws UnirestException, URISyntaxException {
-        storeSubmission();
+        submission = storeSubmission();
```
Comment (ke4) [inline src/test/java/uk/ac/ebi/subs/api/documentation/SubmissionApiDocumentation.java:173]: are you using that variable anywhere?

**Review by ke4 — CHANGES_REQUESTED** [review]
I added some comments. They are only small change requests.

File: src/docs/asciidoc/ref_file_upload.adoc (hunk 2 above)
Comment (haseeb-gh) [reply src/docs/asciidoc/ref_file_upload.adoc:110]: I think its ok not to explain that as its meant to be a typical web link. Average user does not need to know about that. But I did provide a description of this filed below in the CLI section.

File: src/docs/asciidoc/ref_file_upload.adoc (hunk 5 above)
Comment (haseeb-gh) [reply src/docs/asciidoc/ref_file_upload.adoc:149]: no they don't get any notification. it actually doesnt make sense to have files in this directory when all of the submission registered with the share have completed. If there are still files in there at the end then it means that either they were uploaded unintentionally or user decided to remove the file reference from the metadata after uploading the file but forgot to delete the file afterwards.

File: src/main/java/uk/ac/ebi/subs/api/config/MessagingConfig.java (hunk 6 above)
Comment (haseeb-gh) [reply src/main/java/uk/ac/ebi/subs/api/config/MessagingConfig.java:1]: Yes. we are configuring the reply timeout for rabbit sync messaging here. the default timeout is very short and share creation might take slightly longer than that. to be safe, its important to set a relatively longer timeout value.

File: src/docs/asciidoc/ref_file_upload.adoc (hunk 1 above)
Comment (haseeb-gh) [reply src/docs/asciidoc/ref_file_upload.adoc:107]: good idea. please see the changes.

File: src/docs/asciidoc/ref_file_upload.adoc (hunk 3 above)
Comment (haseeb-gh) [reply src/docs/asciidoc/ref_file_upload.adoc:132]: updated.

File: src/docs/asciidoc/ref_file_upload.adoc (hunk 4 above)
Comment (haseeb-gh) [reply src/docs/asciidoc/ref_file_upload.adoc]: updated.

File: src/test/java/uk/ac/ebi/subs/api/documentation/SubmissionApiDocumentation.java (hunk 7 above)
Comment (haseeb-gh) [reply src/test/java/uk/ac/ebi/subs/api/documentation/SubmissionApiDocumentation.java]: thanks for pointing it out. i added that for a new test that i ended up removing in the end anyway but forgot to revert this line. its updated now.

**Review by ke4 — APPROVED** [review]
Thanks for the fixes

**Pull request closed without merging**